openshift/assisted-service
# MGMT-15653: Validate domain in one place

https://issues.redhat.com/browse/MGMT-15653
Uses domain validation function and regex
from validation package for both internal
domain validation and for domain name request
API.

Prevents mismatch in regex validation.


## List all the issues related to this PR


## What environments does this code impact?


## How was this code tested?




E2E KubeAPI test:
1. Create cluster deployment CRs specifying `baseDomain: r--edht.com` & `baseDomain: 000.aa0`
2. Attach agent to cluster deployment
3. Check that domain_name_resolution passes

## Checklist


## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md
